oauth-server/src/main/java/io/strimzi/kafka/oauth/server/JaasServerOauthValidatorCallbackHandler.java

```diff
@@ -0,0 +1,199 @@
+import io.strimzi.kafka.oauth.validator.OAuthIntrospectionValidator;
+import io.strimzi.kafka.oauth.common.TokenInfo;
+import io.strimzi.kafka.oauth.validator.TokenValidator;
+import io.strimzi.kafka.oauth.validator.TokenValidationException;
+import org.apache.kafka.common.errors.AuthenticationException;
+import org.apache.kafka.common.security.auth.AuthenticateCallbackHandler;
+import org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule;
+import org.apache.kafka.common.security.oauthbearer.OAuthBearerToken;
+import org.apache.kafka.common.security.oauthbearer.OAuthBearerValidatorCallback;
+import org.keycloak.jose.jws.JWSInput;
+import org.keycloak.jose.jws.JWSInputException;
+import org.keycloak.representations.AccessToken;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.security.auth.callback.Callback;
+import javax.security.auth.callback.UnsupportedCallbackException;
+import javax.security.auth.login.AppConfigurationEntry;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+import static io.strimzi.kafka.oauth.common.JSONUtil.getClaimFromJWT;
+
+public class JaasServerOauthValidatorCallbackHandler implements AuthenticateCallbackHandler {
+
+    private static final Logger log = LoggerFactory.getLogger(JaasServerOauthValidatorCallbackHandler.class);
+
+    private TokenValidator validator;
+
+    private ServerConfig config;
+
+    private String usernameClaim;
+
+    @Override
+    public void configure(Map<String, ?> configs, String saslMechanism, List<AppConfigurationEntry> jaasConfigEntries) {
+
+        if (!OAuthBearerLoginModule.OAUTHBEARER_MECHANISM.equals(saslMechanism))    {
+            throw new IllegalArgumentException(String.format("Unexpected SASL mechanism: %s", saslMechanism));
+        }
+
+        if (jaasConfigEntries.size() != 1) {
+            throw new IllegalArgumentException("Exactly one jaasConfigEntry expected (size: " + jaasConfigEntries.size());
+        }
+
+        for (AppConfigurationEntry e: jaasConfigEntries) {
+            Properties p = new Properties();
+            p.putAll(e.getOptions());
+            config = new ServerConfig(p);
+            break;
+        }
+
+        validateConfig();
+
+        String jwksUri = config.getValue(ServerConfig.OAUTH_JWKS_ENDPOINT_URI);
+        if (jwksUri != null) {
+
+            validator = new JWTSignatureValidator(
+                    config.getValue(ServerConfig.OAUTH_JWKS_ENDPOINT_URI),
+                    config.getValue(ServerConfig.OAUTH_VALID_ISSUER_URI),
+                    config.getValueAsInt(ServerConfig.OAUTH_JWKS_EXPIRY_SECONDS, 300),
+                    config.getValueAsInt(ServerConfig.OAUTH_JWKS_REFRESH_SECONDS, 360),
+                    config.getValueAsBoolean(ServerConfig.OAUTH_VALIDATE_COMMON_CHECKS, true),
```

Comment:
Do we support this value? If not, we should remove it from here.

Reply:
Yeah, I think common checks should always be on. I'll remove option.